client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/propagation/jax/JMSMessageExtractAdapter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2015-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.agent.opentracing.propagation.jax;
```

Comment:
Any suggestions? :) Trying to scope the classes supporting propagation to the category of rules, but alternative could be to just put them all at the same level? (i.e. ...agent.opentracing.propagation)